ui/ducks/send/send.js

```diff
@@ -3013,35 +3012,36 @@ export function signTransaction(history) {
         };
 
         if (bestQuote?.approvalNeeded) {
-          const { id } = await addTransactionAndWaitForPublish(
-            { ...bestQuote.approvalNeeded, amount: '0x0' },
-            {
-              requireApproval: false,
-              // TODO: create new type for swap+send approvals; works as stopgap bc swaps doesn't use this type for STXs in `submitSmartTransactionHook` (via `TransactionController`)
-              type: TransactionType.swapApproval,
-              swaps: {
-                hasApproveTx: true,
-                meta: {
-                  type: TransactionType.swapApproval,
-                  sourceTokenSymbol,
+          const { id } = await dispatch(
+            addTransactionAndRouteToConfirmationPage(
```

Comment:
I want to stress I think this PR is a great improvement, since we're now centralising the transaction flows through a consistent code path, confirmation, and UX.

But as a point of interest and future discussion, the send flow architecture, including the associated duck, Redux state, and draft transaction, all introduce a lot of complexity on top of the transaction code paths that can feel more a side effect of legacy code as opposed to intentional design.

If we were to implement wallet transfers for the first time today, it feels it would be far simpler, modular, and maintainable, if we were to think of a transfer as fundamentally just a standard transaction confirmation but with some additional modular React components to specify tokens and networks and amounts, and update the underlying data.

As a result, we could replace lots of the large send logic like this function, with many modular, typed and functional React components and hooks, and use them to update a pending transaction directly in the `TransactionController` state. This would mean the existing Redux state could be replaced by the React components state and the `TransactionController` unapproved transaction.

This would also align with the ongoing confirmation re-design which is based around a single confirmation template with variations, custom inputs and rows dependent on the type.